chore: upgrade terraform to 1.10.5

coadler opened the pull request:
Updates `terraform` to [v1.10.5](https://github.com/hashicorp/terraform/blob/v1.10.5/CHANGELOG.md#1105-january-22-2025)

coadler committed:
chore: upgrade terraform to 1.10.5

coadler committed:
fixup! chore: upgrade terraform to 1.10.5

matifali reviewed (approved):
LGTM except the testdata changes.

matifali commented on provisioner/terraform/testdata/chaining-resources/chaining-resources.tfplan.json:
```diff
@@ -79,29 +80,29 @@
           "connection_timeout": 120,
           "dir": null,
           "env": null,
+          "login_before_ready": true,
```
I think we removed these in #16309 and should not be added back.

coadler replied:
These are auto-gen'd by terraform, I can't control if they're re-added or not. Were they removed manually before?

coadler replied:
The PR you linked is slightly confusing. I don't see anywhere that it's updating the provider versions

johnstcn replied:
@defelmnq can you double-check what's going on here?

matifali replied:
Also, we still have `v1.0.4` of the provider in `go.mod`. So this is confusing me, too. I was of the impression we updated the provider to v2. 
https://github.com/coder/coder/blob/1ce4dfe0583cd74356d7dde8215f95e20a527b02/go.mod#L97

SasSwart replied:
I think it might be related to these log lines when I run generate.sh:

```
╷
│ Error: Unsupported block type
│
│   on multiple-agents-multiple-monitors.tf line 13, in resource "coder_agent" "dev1":
│   13:   resources_monitoring {
│
│ Blocks of type "resources_monitoring" are not expected here.
╵
╷
│ Error: Unsupported block type
│
│   on multiple-agents-multiple-monitors.tf line 24, in resource "coder_agent" "dev2":
│   24:   resources_monitoring {
│
│ Blocks of type "resources_monitoring" are not expected here.
```

matifali replied:
`resources_monitoring` was added into `main` and has not been tagged yet.

johnstcn reviewed (dismissed):
Agree with @matifali , not sure about those testdata changes. The rest LGTM.

johnstcn committed:
chore(provisioner/terraform/testdata): update provider versions, fix resource monitor tests (#16601)

Updates to https://github.com/coder/coder/pull/16519

- Fixes missing leading `/` in volumes in resource monitor tests
- Updates provider to `>=2.0.0` in provider testdata
- Fixes provider to required release version for resource monitors

matifali commented:
Should rebase on `main` after #16601 is merged.

johnstcn commented:
> Should rebase on `main` after #16601 is merged.

I merged it into this PR!

johnstcn reviewed (approved):
(semi-biased approval alert)

stirby commented:
/cherry-pick release/2.19
